packages/medusa/src/types/payment-processor.ts

```diff
@@ -0,0 +1,26 @@
+import { Address, Customer, ShippingMethod } from "../models"
+
+export type PaymentProcessorContext = {
+  cart?: {
+    context: Record<string, unknown>
+    id: string
+    email: string
+    shipping_address: Address | null
```

Comment:
suggestion: Add support for `billing_address`

Reply:
I was thinking, do we really need the cart data for the payment processor? Or would it be more for the checkout API?

Reply:
No, as this will be an entirely new interface, I think it would be best to remove the need for a cart in the context of payments and instead rely on a set of required props